### PR TITLE
feature/jl/fix_SqlExecutorTest

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-db/src/test/java/org/sonar/db/DbTester.java [hunk 1]
```diff
@@ -145,12 +147,13 @@ public void executeInsert(String table, Map<String, String> valuesByColumn){
     if (valuesByColumn.isEmpty()) {
       throw new IllegalArgumentException("Values cannot be empty");
     }
+
     String sql = "insert into " + table.toLowerCase() + " (" +
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/DbTester.java]: I'm faster than SQ analysis : `toLowerCase(Locale.ENGLISH)`

File: sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java [hunk 2]
```diff
@@ -73,13 +60,12 @@ public void executeSelect_executes_PreparedStatement() throws Exception {
   @Test
   public void executeUpdate_executes_PreparedStatement() throws Exception {
     dbTester.executeInsert(USERS_DB_TABLE, ImmutableMap.of(LOGIN_DB_COLUMN, "the_login", NAME_DB_COLUMN, "the name"));
-    dbTester.commit();
 
     try (Connection connection = dbTester.openConnection()) {
       underTest.executeUpdate(connection, "update users set " + NAME_DB_COLUMN + "='new name' where " + LOGIN_DB_COLUMN + "='the_login'");
-      connection.commit();
     }
-    Map<String, Object> row = dbTester.selectFirst("select " + NAME_DB_COLUMN + " from users where " + LOGIN_DB_COLUMN + "='the_login'");
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java]: this fix is not enough for mysql as the returned key is `name` but not `NAME` (see line 69). Not a problem so far, I'm improving DbTester#selectFirst() in another branch.

File: sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java [hunk 3]
```diff
@@ -44,21 +41,11 @@
   @Rule
   public DbTester dbTester = DbTester.create(System2.INSTANCE);
 
-  @Before
-  public void disableIfNotH2() {
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java]: This fix is not enough for supporting all dbs. DbTester#selectFirst() must be fixed for supporting case-sensitive column names. This is part of another branch (PR pushed soon). Meanwhile assumption on H2 should be kept.

Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java]: Oh, I missed that. Note that in order tests the aliases are quoted (`'NAME'`)

Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/charset/SqlExecutorTest.java]: ok, I'll ad quotes

**Comment by julienlancelot** [comment]
@simonbrandhof Feedbacks applied

**Comment by simonbrandhof** [comment]
LGTM

**Pull request merged**